docs/custom-domain/registry-setup.md

```diff
@@ -0,0 +1,95 @@
+# Using Custom Domains with Azure Container Registry
+
+Every ACR is accessed using its login server. If you have a registry called `myregistry`, you access it using its default hostname, `myregistry.azurecr.io` (in Azure Public Cloud.) As a customer belonging to an organization, you may prefer to access your registry using a custom domain that is associated with your organization, for instance, `container-registry.contoso.com`.
+
+The following steps describe how you can achieve this.
+
+## Prerequisites
+- [Azure CLI](https://docs.microsoft.com/cli/azure/?view=azure-cli-latest): version 2.4.0 or higher
+  - Consider using [Azure Cloud Shell](https://docs.microsoft.com/azure/cloud-shell/overview)
+- A _premium_ Azure Container Registry. See [here](https://docs.microsoft.com/azure/container-registry/container-registry-get-started-azure-cli) for instructions on how to create one.
+- Your custom domain names. The following two are required:
+  - Custom registry domain to access the registry REST API. Example for the `contoso.com` domain: `container-registry.contoso.com` 
+  - Custom data domain to access the registry content. Again, example for `contoso.com`: `eastus-registry-data.contoso.com`
+    - Note that the custom data domain is region specific. For geo-replicated registries, each region should have its own custom data endpoint.
+
+  For each domain, you must prepare a single PEM formatted file containing the TLS private key and public certificate:
+  
+  ```
+  -----BEGIN PRIVATE KEY-----  
+  .....  
+  -----END PRIVATE KEY-----  
+  -----BEGIN CERTIFICATE-----  
+  .....  
+  -----END CERTIFICATE-----
+  ```
+  
+  For example, using [openssl](https://github.com/openssl/openssl):
+  - Create a self-signed public cert and private key
+     - `openssl req -nodes -x509 -newkey rsa:4096 -keyout container-registry.contoso.com.key.pem -out container-registry.contoso.com.cert.pem -days 365 -subj '/CN=container-registry.contoso.com/O=Contoso./C=US'`
+  - Create a single file containing both the public certificate and private key
+     - `cat container-registry.contoso.com.key.pem >> container-registry-contoso-com-pem`
+     - `cat container-registry.contoso.com.cert.pem >> container-registry-contoso-com-pem`
+  - For each data domain, follow the same steps above to prepare the PEM formatted files containing the public certificate and private key.
+  
+  Azure Key Vault allows you to [create](https://docs.microsoft.com/azure/key-vault/certificate-scenarios) Certificate Authority (CA) signed certificates.
+  
+## Prepare your existing registry
+We will enable two features on your registry that are currently in preview:
+- Data Endpoints:\
+  This feature provides a dedicated endpoint for downloading content from your registry. If you have a registry in East US, on enabling this feature, a data endpoint is automatically created for you: `eastus.data.myregistry.azurecr.io`
```

Comment:
I created data endpoints for a registry and this is what I got:

  "dataEndpointHostNames": [
    "[registryname].[region].data.azurecr.io"
  ]

This suggests it would be [region].data.[registryname].azurecr.io

Reply:
Addressed